src/plugins/navigation_embeddable/public/components/navigation_embeddable_component.tsx

```diff
@@ -8,41 +8,53 @@
 
 import React, { useMemo } from 'react';
 
-import { EuiPanel } from '@elastic/eui';
+import { EuiListGroup, EuiPanel } from '@elastic/eui';
 
-import { DASHBOARD_LINK_TYPE } from '../embeddable/types';
 import { useNavigationEmbeddable } from '../embeddable/navigation_embeddable';
 import { ExternalLinkComponent } from './external_link/external_link_component';
+import { DASHBOARD_LINK_TYPE, NAV_HORIZONTAL_LAYOUT } from '../embeddable/types';
 import { DashboardLinkComponent } from './dashboard_link/dashboard_link_component';
 import { memoizedGetOrderedLinkList } from '../editor/navigation_embeddable_editor_tools';
 
+import './navigation_embeddable_component.scss';
+
 export const NavigationEmbeddableComponent = () => {
   const navEmbeddable = useNavigationEmbeddable();
 
   const links = navEmbeddable.select((state) => state.explicitInput.links);
+  const layout = navEmbeddable.select((state) => state.explicitInput.layout);
   const orderedLinks = useMemo(() => {
     return memoizedGetOrderedLinkList(links);
   }, [links]);
 
-  /** TODO: Render this as a list **or** "tabs" as part of https://github.com/elastic/kibana/issues/154357 */
-  return (
-    <EuiPanel className="eui-yScroll">
-      {orderedLinks.map((link) => {
-        return (
-          <EuiPanel
-            id={`navigationLink--${link.id}`}
-            key={`${link.id}`}
-            paddingSize="none"
-            hasShadow={false}
-          >
-            {link.type === DASHBOARD_LINK_TYPE ? (
-              <DashboardLinkComponent link={link} />
+  const linkItems: { [id: string]: { id: string; content: JSX.Element } } = useMemo(() => {
+    return Object.keys(links).reduce((prev, currentLinkId) => {
+      const currentLink = links[currentLinkId];
+      return {
+        ...prev,
+        [currentLink.id]: {
+          id: currentLink.id,
+          content:
+            currentLink.type === DASHBOARD_LINK_TYPE ? (
+              <DashboardLinkComponent key={currentLink.id} link={currentLink} />
             ) : (
-              <ExternalLinkComponent link={link} />
-            )}
-          </EuiPanel>
-        );
-      })}
+              <ExternalLinkComponent key={currentLink.id} link={currentLink} />
+            ),
+        },
+      };
+    }, {});
+  }, [links]);
+
+  return (
+    <EuiPanel
+      className={`navEmbeddableComponent ${
+        layout === NAV_HORIZONTAL_LAYOUT ? 'eui-xScroll' : 'eui-yScroll'
+      }`}
+      paddingSize="xs"
+    >
+      <EuiListGroup maxWidth={false} className={`${layout}LayoutWrapper`}>
+        {orderedLinks.map((link) => linkItems[link.id].content)}
+      </EuiListGroup>
```

Comment:
I opted to render the list **always** as an `EuiListGroup` so that the related HTML would be consistent - if I used the `EuiTabs` component for the horizontal layout, the HTML underneath renders as a bunch of buttons (**without** the surrounding `ul` and `li` tags that `EuiListGroup` includes) with the `tabs` aria-role attribute. 

This is not great from an accessibility standpoint, since these links are **not** actually functioning as tabs and are **actually** functioning as more of a menu (i.e. they navigate to a different URL rather than simply loading new content on the current page) - so, using the `tabs` aria-role is misleading in this case.